locale/pt-br/knowledge/javascript-conventions/what-is-the-arguments-object.md

```diff
@@ -0,0 +1,94 @@
+---
+date: '2011-08-26T10:08:50.000Z'
+tags:
+  - truthy
+  - falsy
+  - types
+  - coercion
+title: What is the arguments object?
+difficulty: 4
+layout: knowledge-post.hbs
+---
+
+Os objetos `arguments` são uma estrutura especial disponível dentro de todas as chamadas de função e que é representada como uma lista de argumentos que são passadas quando a função é chamada. Desde que JavaScript permite chamar funções com indeterminado número de argumentos, nós precisamos de um método para descobrir e acessar esses argumentos.
+
+Os objetos `arguments` são um tipo de objeto array, seu tamanho corresponde a quantidade de argumentos passados para a função e que é possível acessar seu valor através de indexação do Array. Exemplo `arguments[0]` //Captura ao primeiro argumento. Existe apenas mais uma propriedade de `arguments` chamada `callee`, Nota: o ES5 impede o uso do `strict mode` mais informações podem ser encontradas [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments/callee). Veja mais exemplos sobre as propriedades `arguments`.
+
+```js
+const minhaFunc = function(primeiro) {
+  const index0 = arguments[0] === primeiro;
+  const index1 = arguments[1] === 2;
+  const argumentsLength = arguments.length === 3;
+  return console.log(index0, index1, argumentsLength)
+}
+minhaFunc(1, 2, 3);
+// expectativa de retorno no console: true | true | true
+```
```

Comment:
In between those two examples, there is another paragraph which is missing:

*"This construct is very useful and gives javascript functions a lot of flexibility. But there is an important gotcha. The arguments object behaves like an array, but it is not an actual array. It does not have Array in its prototype chain and it does not respond to any array methods, e.g. arguments.sort() raises a TypeError. Instead, you need to copy the values into a true array first. With the advent of ES6 Array.from() method this is quite straightforward."*

Please add it so the docs are complete :)